### PR TITLE
Fix error messages and typo

### PR DESCRIPTION
**Pull request by Shairam**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -287,11 +291,12 @@ service studentData on studentServiceListener {
         if (result is error) {
             log:printError("Error sending response", err = result);
         }
-        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
         _ = observe:addTagToSpan("tot_requests", <string>studentData.requestCounts);
         _ = observe:addTagToSpan("error_counts", <string>studentData.errors);
     }
 
+    // Resource configuration for getting marksa of a particular student.
```
Comment (Anoukh) [inline README.md]:
```suggestion
    // Resource configuration for getting marks of a particular student.
```

File: README.md [hunk 2]
```diff
@@ -461,20 +468,20 @@ type Marks record {
...
 }
-
 service MarksData on marksServiceListener {
+    // Resource configuration for getting student's marks details from the system.
     @http:ResourceConfig {
         methods: ["GET"],
         path: "/getMarks/{stuId}"
     }
-    // Get marks resource used to get student's marks.
+    // Get marks is a  resource used to get student's marks.
```
Comment (Anoukh) [inline README.md]:
```suggestion
    // Resource used to get student's marks.
```

File: README.md [hunk 3]
```diff
@@ -594,26 +597,26 @@ public function main() {
...
+        log:printError("Error in checking int type of " + input, err = isInt);
         return false;
     } else {
         return isInt;
     }
 
 }
 
-// Function  to add students to database.
+// Function  to add students to the database.
```
Comment (Anoukh) [inline README.md]:
```suggestion
// Function to add students to the database.
```

File: README.md [hunk 4]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
+            log:printError("Error in extracting JSON from response", err = jsonMsg);
         }
 
     } else {
-        log:printError("Error in response", err = resp);
+        log:printError("Error in obtained response", err = resp);
     }
 }
 
+// Function  to view all student's details to the database.
```
Comment (Anoukh) [inline README.md]:
```suggestion
// Function to view all student's details
```

File: guide/client_service/client_main.bal [hunk 5]
```diff
@@ -91,33 +90,33 @@ function addStudent(http:Request req) {
...
-        io:println("Adding students failed");
+        log:printError("Error in converting age and mobile number to int", err = ageAsInt);
         return;
     }
 
     // Send the request to students service and get the response from it.
     var resp = studentService->post("/records/addStudent", req);
 
     if (resp is http:Response) {
+        // Extracting data from JSON received.
```
Comment (Anoukh) [inline guide/client_service/client_main.bal]:
```suggestion
        // Extracting data from received JSON.
```

File: README.md [hunk 6]
```diff
@@ -141,18 +141,18 @@ mysql:Client studentDB = new({
         dbOptions: { useSSL: false }
     });
 
-// Listener for the student service.
+// Port listener for the student service.
```
Comment (praneesha) [inline README.md]: Listener of the student service port.

File: README.md [hunk 7]
```diff
@@ -141,18 +141,18 @@ mysql:Client studentDB = new({
...
 listener http:Listener studentServiceListener = new(9292);
 
-// Student data service.
+// Service configuration for student data service.
```
Comment (praneesha) [inline README.md]: Service configuration for the student data service.

File: README.md [hunk 8]
```diff
@@ -168,29 +168,32 @@ service studentData on studentServiceListener {
...
-
             Student|error studentDetails = Student.convert(payloadJson);
 
             if (studentDetails is Student) {
                 io:println(studentDetails);
                 // Calling the function insertData to update database.
                 json returnValue = insertData(untaint studentDetails.name, untaint studentDetails.age, untaint studentDetails.mobNo, untaint studentDetails.address);
                 response.setJsonPayload(untaint returnValue);
+            } else {
+                log:printError("Error in converting JSON payload to student type", err = studentDetails);
```
Comment (praneesha) [inline README.md]: Error in converting the JSON payload to the 'student' type.

File: README.md [hunk 9]
```diff
@@ -168,29 +168,32 @@ service studentData on studentServiceListener {
...
             }
+        } else {
+            log:printError("Error obtaining the JSON payload", err = payloadJson);
```
Comment (praneesha) [inline README.md]: Error in obtaining the JSON payload.

File: README.md [hunk 10]
```diff
@@ -168,29 +168,32 @@ service studentData on studentServiceListener {
...
         }
-
-        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
```
Comment (praneesha) [inline README.md]: The below function adds tags to be passed as metrics in the traces. These tags are added to the default system span.

File: README.md [hunk 11]
```diff
@@ -168,29 +168,32 @@ service studentData on studentServiceListener {
...
         _ = observe:addTagToSpan("tot_requests", <string>studentData.requestCounts);
         _ = observe:addTagToSpan("error_counts", <string>studentData.errors);
 
         // Send the response back to the client with the returned JSON value from insertData function.
         var result = caller->respond(response);
         if (result is error) {
             // Log the error for the service maintainers.
-            log:printError("Error responding to the client", err = result);
+            log:printError("Error sending response", err = result);
```
Comment (praneesha) [inline README.md]: Error in sending the response to the client.

File: README.md [hunk 12]
```diff
@@ -168,29 +168,32 @@ service studentData on studentServiceListener {
...
         }
     }
 
+    // Resource configuration for viewing all the student's details in the system.
```
Comment (praneesha) [inline README.md]: Resource configuration for viewing details of all the students in the system.

File: README.md [hunk 13]
```diff
@@ -201,16 +204,17 @@ service studentData on studentServiceListener {
...
+
         //Sending a request to MySQL endpoint and getting a response with required data table.
         var returnValue = studentDB->select("SELECT * FROM student", Student, loadToMemory = true);
         _ = observe:finishSpan(spanId2);
+
         // A table is declared with Student as its type.
         table<Student> dataTable = table{};
 
         if (returnValue is error) {
-            io:println("Select data from student table failed");
```
Comment (praneesha) [inline README.md]: Error in selecting data from the student table.

File: README.md [hunk 14]
```diff
@@ -201,16 +204,17 @@ service studentData on studentServiceListener {
...
+           log:printError("Error in fetching students data from the database", err = returnValue);
```
Comment (praneesha) [inline README.md]: Error in fetching details of all the students from the database.

File: README.md [hunk 15]
```diff
@@ -223,27 +227,28 @@ service studentData on studentServiceListener {
         // Table is converted to JSON.
         var jsonConversionValue = json.convert(dataTable);
         if (jsonConversionValue is error) {
-            status = { "Status": "Data Not available" };
+            log:printError("Error in converting the data table from database to JSON", err = jsonConversionValue);
```
Comment (praneesha) [inline README.md]: Error in converting the data from the tabular format to JSON.

File: README.md [hunk 16]
```diff
@@ -223,27 +227,28 @@ service studentData on studentServiceListener {
...
         } else {
             status = jsonConversionValue;
         }
         // Sending back the converted JSON data to the request made to this service.
         response.setJsonPayload(untaint status);
         var result = caller->respond(response);
         if (result is error) {
-            log:printError("Error sending response", err = result);
+            log:printError("Error in sending response", err = result);
```
Comment (praneesha) [inline README.md]: Error in sending the response to the client.

File: README.md [hunk 17]
```diff
@@ -223,27 +227,28 @@ service studentData on studentServiceListener {
...
         }
 
         if (childSpanId is int) {
             _ = observe:finishSpan(childSpanId);
         } else {
             log:printError("Error attaching span ", err = childSpanId);
         }
-        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
```
Comment (praneesha) [inline README.md]: The below function adds tags to be passed as metrics in the traces. These tags are added to the default system span.

File: README.md [hunk 18]
```diff
@@ -223,27 +227,28 @@ service studentData on studentServiceListener {
...
         _ = observe:addTagToSpan("tot_requests", <string>studentData.requestCounts);
         _ = observe:addTagToSpan("error_counts", <string>studentData.errors);
     }
 
+    // Resource configuration for making a mock error to the system.
```
Comment (praneesha) [inline README.md]: Resource configuration for generating a mock error in the system for observability purposes.

File: README.md [hunk 19]
```diff
@@ -252,10 +257,9 @@ service studentData on studentServiceListener {
     resource function testError(http:Caller caller, http:Request request) {
         studentData.requestCounts += 1;
         http:Response response = new;
-
         studentData.errors += 1;
         io:println(studentData.errors);
-        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
```
Comment (praneesha) [inline README.md]: The below function adds tags to be passed as metrics in the traces. These tags are added to the default system span.

File: README.md [hunk 20]
```diff
@@ -266,11 +270,11 @@ service studentData on studentServiceListener {
         }
     }
 
+    // Resource configuration for deleting a students detail's from the system.
```
Comment (praneesha) [inline README.md]: Resource configuration for deleting details of a student from the system.

File: README.md [hunk 21]
```diff
@@ -287,11 +291,12 @@ service studentData on studentServiceListener {
         if (result is error) {
             log:printError("Error sending response", err = result);
         }
-        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
```
Comment (praneesha) [inline README.md]: The below function adds tags to be passed as metrics in the traces. These tags are added to the default system span.

File: README.md (hunk 1 above)
Comment (praneesha) [inline README.md]: Resource configuration for retrieving the marks of a particular student.

File: README.md [hunk 22]
```diff
@@ -306,13 +311,12 @@ service studentData on studentServiceListener {
         int|error firstSpan = observe:startSpan("First span");
         // Request made for obtaining marks of the student with the respective studentId to marks Service.
         var requestReturn = marksServiceEP->get("/marks/getMarks/" + untaint studentId);
-
         if (requestReturn is error) {
-            log:printError("Error", err = requestReturn);
+            log:printError("Error in fetching marks from the marks system", err = requestReturn);
```
Comment (praneesha) [inline README.md]: Error in fetching the marks from the system.

File: README.md [hunk 23]
```diff
@@ -306,13 +311,12 @@ service studentData on studentServiceListener {
...
         } else {
             var msg = requestReturn.getJsonPayload();
             if (msg is error) {
-                log:printError("Error", err = msg);
+                log:printError("Error in extracting JSON from the response", err = msg);
```
Comment (praneesha) [inline README.md]: Error in extracting the JSON payload from the response.

File: README.md [hunk 24]
```diff
@@ -330,7 +334,7 @@ service studentData on studentServiceListener {
         if (resResult is error) {
             log:printError("Error sending response", err = resResult);
         }
-        //  The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
+        //  The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default system span.
```
Comment (praneesha) [inline README.md]: The below function adds tags to be passed as metrics in the traces. These tags are added to the default system span.

File: README.md [hunk 25]
```diff
@@ -350,17 +354,17 @@ public function insertData(string name, int age, int mobNo, string address) retu
...
-
     if (returnValue is int) {
         table<Student> result = getId(untaint mobNo);
         while (result.hasNext()) {
             var returnValue2 = result.getNext();
             if (returnValue2 is Student) {
                 uniqueId = returnValue2.id;
             } else {
-                io:println("Unable to get student details ");
+                log:printError("Error in obtaining student ID from the database", err = returnValue2);
```
Comment (praneesha) [inline README.md]: Error in obtaining the student ID from the database for the added student.

File: README.md [hunk 26]
```diff
@@ -369,6 +373,8 @@ public function insertData(string name, int age, int mobNo, string address) retu
         } else {
             updateStatus = { "Status": "Data Not inserted" };
         }
+    } else {
+        log:printError("Error in adding the data to the database", err = returnValue);
```
Comment (praneesha) [inline README.md]: Error in adding the data to the database.

File: README.md [hunk 27]
```diff
@@ -396,6 +402,7 @@ public function deleteData(int studentId) returns (json) {
         }
 
     } else {
+        log:printError("Error in removing data from the database", err = returnValue);
```
Comment (praneesha) [inline README.md]: Error in removing the data from the database.

File: README.md [hunk 28]
```diff
@@ -412,13 +419,13 @@ public function getId(int mobNo) returns table<Student> {
...
     string sqlString = "SELECT * FROM student WHERE mobNo = ?";
     // Retrieve student data by invoking select remote function defined in ballerina sql client
-    var ret = studentDB->select(sqlString, Student, mobNo);
+    var returnValue = studentDB->select(sqlString, Student, mobNo);
 
     table<Student> dataTable = table{};
-    if (ret is error) {
-        log:printError("Error ", err = ret);
+    if (returnValue is error) {
+        log:printError("Error obtaining student ID from the database", err = returnValue);
```
Comment (praneesha) [inline README.md]: Error in obtaining a student ID from the database for the added student.

File: README.md [hunk 29]
```diff
@@ -461,20 +468,20 @@ type Marks record {
     int science;
 };
 
-// Listener for marks service.
+// Port listener for marks service.
```
Comment (praneesha) [inline README.md]: Listener for the port of the marks service.

File: README.md [hunk 30]
```diff
@@ -461,20 +468,20 @@ type Marks record {
...
 listener http:Listener marksServiceListener = new(9191);
 
-// Marks data service.
+// Service configuration marks service.
```
Comment (praneesha) [inline README.md]: Service configuration of the marks service.

File: README.md [hunk 31]
```diff
@@ -461,20 +468,20 @@ type Marks record {
...
 @http:ServiceConfig {
     basePath: "/marks"
 }
-
 service MarksData on marksServiceListener {
+    // Resource configuration for getting student's marks details from the system.
```
Comment (praneesha) [inline README.md]: Resource configuration for retrieving marks of a student from the system.

File: README.md [hunk 32]
```diff
@@ -461,20 +468,20 @@ type Marks record {
...
     @http:ResourceConfig {
         methods: ["GET"],
         path: "/getMarks/{stuId}"
     }
-    // Get marks resource used to get student's marks.
+    // Resource used to get student's marks.
```
Comment (praneesha) [inline README.md]: Resource to retrieve the marks of a student.

File: README.md [hunk 33]
```diff
@@ -499,14 +506,12 @@ public function findMarks(int stuId) returns (json) {
...
     var returnValue = studentDB->select(sqlString, Marks, loadToMemory = true);
-
     // Assigning data obtained from db to a table.
     table<Marks> dataTable = table {};
-
     if (returnValue is table<Marks>) {
         dataTable = returnValue;
     } else {
-        log:printError("Error Detected", err = returnValue);
+        log:printError("Error in fetching data table from the database", err = returnValue);
```
Comment (praneesha) [inline README.md]: Error in fetching the data table from the database.

File: README.md [hunk 34]
```diff
@@ -517,9 +522,8 @@ public function findMarks(int stuId) returns (json) {
         status = jsonConversionValue;
     } else {
         status = { "Status": "Data Not available" };
-        log:printError("Error Detected", err = jsonConversionValue);
+        log:printError("Error in converting the data table to JSON", err = jsonConversionValue);
```
Comment (praneesha) [inline README.md]: Error in converting the data from the tabular format to JSON.

File: README.md [hunk 35]
```diff
@@ -550,6 +554,7 @@ import ballerina/http;
 import ballerina/io;
 import ballerina/log;
 
+// Host name for the student administration system server.
```
Comment (praneesha) [inline README.md]: Hostname for the server hosting the student administration system.

File: README.md [hunk 36]
```diff
@@ -572,17 +577,15 @@ public function main() {
...
             continue;
         }
-
         var intOperation = int.convert(choice);
-
         if (intOperation is int) {
             io:println(intOperation);
             operation = intOperation;
+        } else {
+            log:printError("Error in converting user selected option to int type", err = intOperation);
```
Comment (praneesha) [inline README.md]: Error in converting the option selected by the user to an integer.

File: README.md [hunk 37]
```diff
@@ -594,26 +597,26 @@ public function main() {
     }
 }
 
+// Function to check if an input is integer
```
Comment (praneesha) [inline README.md]: Function to check if an input is an integer.

File: README.md [hunk 38]
```diff
@@ -594,26 +597,26 @@ public function main() {
...
 function isInteger(string input) returns boolean {
     string regEx = "\\d+";
     boolean|error isInt = input.matches(regEx);
     if (isInt is error) {
-        log:printError("Error", err = isInt);
+        log:printError("Error in checking int type of " + input, err = isInt);
```
Comment (praneesha) [inline README.md]: Error in checking if the 'input' variable is an integer.

File: README.md [hunk 39]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
         req.setJsonPayload(jsonMsg);
 
     } else {
-        io:println("Adding students failed");
```
Comment (praneesha) [inline README.md]: Error in adding student details.

File: README.md [hunk 40]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
+        log:printError("Error in converting age and mobile number to int", err = ageAsInt);
```
Comment (praneesha) [inline README.md]: Error in converting the age and the mobile number to integers.

File: README.md [hunk 41]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
         return;
     }
 
     // Send the request to students service and get the response from it.
     var resp = studentService->post("/records/addStudent", req);
 
     if (resp is http:Response) {
+        // Extracting data from received JSON.
```
Comment (praneesha) [inline README.md]: Extracting data from the received JSON object.

File: README.md [hunk 42]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
         var jsonMsg = resp.getJsonPayload();
         if (jsonMsg is json) {
             string message = "Status: " + jsonMsg["Status"] .toString() + " Added Student Id :- " +
                 jsonMsg["id"].toString();
-            // Extracting data from JSON received and displaying.
             io:println(message);
         } else {
-            log:printError("Error in JSON", err = jsonMsg);
+            log:printError("Error in extracting JSON from response", err = jsonMsg);
```
Comment (praneesha) [inline README.md]: Error in extracting the JSON payload from the received response.

File: README.md [hunk 43]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
         }
 
     } else {
-        log:printError("Error in response", err = resp);
+        log:printError("Error in obtained response", err = resp);
```
Comment (praneesha) [inline README.md]: Error in the obtained response.

File: README.md [hunk 44]
```diff
@@ -623,33 +626,33 @@ function addStudent(http:Request req) {
...
     }
 }
 
+// Function to view all student's details.
```
Comment (praneesha) [inline README.md]: Function to view details of all the students.

File: README.md [hunk 45]
```diff
@@ -673,14 +676,15 @@ function viewAllStudents() {
             }
 
         } else {
-            log:printError("Error ", err = jsonMsg);
+            log:printError("Error in extracting JSON from response", err = jsonMsg);
```
Comment (praneesha) [inline README.md]: Error in extracting the JSON payload from the received response.

File: README.md [hunk 46]
```diff
@@ -673,14 +676,15 @@ function viewAllStudents() {
...
         }
 
     } else {
-        log:printError("Error ", err = response);
+        log:printError("Error in obtained response", err = resp);
```
Comment (praneesha) [inline README.md]: Error in the obtained response.

File: README.md [hunk 47]
```diff
@@ -673,14 +676,15 @@ function viewAllStudents() {
...
     }
 }
 
+// Function to delete a student's data from the system.
```
Comment (praneesha) [inline README.md]: Function to delete the details of a student from the system

File: README.md [hunk 48]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
             string message = jsonMsg["Status"].toString();
             io:println("\n" + message + "\n");
         } else {
-            log:printError("Error ", err = jsonMsg);
+            log:printError("Error in extracting JSON from response", err = jsonMsg);
```
Comment (praneesha) [inline README.md]: Error in extracting the JSON payload from the received response.

File: README.md [hunk 49]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
         }
+    } else {
+        log:printError("Error in obtained response", err = resp);
```
Comment (praneesha) [inline README.md]: Error in the obtained response.

File: README.md [hunk 50]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
     }
 
 }
 
+// Function to make a mock error in the system for observability purpose.
```
Comment (praneesha) [inline README.md]: Function to generate a mock error in the system for observability purposes.

File: README.md [hunk 51]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
     var response = studentService->get("/records/testError");
-
     if (response is http:Response) {
         var msg = response.getTextPayload();
-
         if (msg is string) {
             io:println("\n" + msg + "\n");
         } else {
-            log:printError("Error", err = msg);
+            log:printError("Error in fetching text from response", err = msg);
```
Comment (praneesha) [inline README.md]: Error in fetching the text from the received response.

File: README.md [hunk 52]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
         }
     } else {
-        log:printError("Error", err = response);
+        log:printError("Error in obtained response", err = response);
```
Comment (praneesha) [inline README.md]: Error in the obtained response.

File: README.md [hunk 53]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
     }
 }
 
+// Function to fetch marks details of a student from the system.
```
Comment (praneesha) [inline README.md]: Function to fetch the marks of a student from the system.

File: README.md [hunk 54]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
 function getMarks() {
-    // Get student id.
+    // Get student ID.
```
Comment (praneesha) [inline README.md]: Get the student ID of a particular student.

File: README.md [hunk 55]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
                     +
                     " Science: " + jsonMsg[0]["science"] .toString();
-
             } else {
                 message = "Data not available. Check if student's mark is added or student might not be in our system.";
             }
             io:println("\n" + message + "\n");
         } else {
-            log:printError("Error ", err = jsonMsg);
+            log:printError("Error in extracting JSON from response", err = jsonMsg);
```
Comment (praneesha) [inline README.md]: Error in extracting the JSON payload from the response.

File: README.md [hunk 56]
```diff
@@ -694,54 +698,53 @@ function deleteStudent() {
...
         }
     } else {
-        log:printError("Error ", err = response);
+        log:printError("Error in obtained response ", err = response);
```
Comment (praneesha) [inline README.md]: Error in the obtained response.

**Review by praneesha — CHANGES_REQUESTED** [review]
Added a few suggestions.

**Review by praneesha — APPROVED** [review]

**Pull request merged**